File: src/runtime-helpers/cronium.js
const fs = require("fs");

class Cronium {
  constructor() {
    // Load input data automatically
    try {
      this._inputData = JSON.parse(fs.readFileSync("input.json", "utf8"));
    } catch (error) {
      this._inputData = {};
    }

    // Load event data automatically
    try {
      this._eventData = JSON.parse(fs.readFileSync("event.json", "utf8"));
    } catch (error) {
      this._eventData = {};
    }

    // Initialize condition state
    this._condition = null;
  }

  input() {
    return this._inputData;
  }

  output(data) {
    try {
      fs.writeFileSync("output.json", JSON.stringify(data, null, 2));
    } catch (error) {
      console.error("Error writing output:", error.message);
    }
  }

  event() {
    return this._eventData;
  }

  setCondition(condition) {
    // Convert the condition to a boolean and store it
    this._condition = Boolean(condition);

    // Write the condition result to a file for the executor to read
    try {
      fs.writeFileSync(
        "condition.json",
        JSON.stringify({ condition: this._condition }, null, 2),
      );
    } catch (error) {
      console.error("Error writing condition:", error.message);
    }

    return this._condition;
  }

  getCondition() {
    return this._condition;
  }

  getVariable(key) {
    try {
      const variablesData = JSON.parse(
        fs.readFileSync("variables.json", "utf8"),
      );
      return variablesData[key] || null;
    } catch (error) {
      return null;
    }
  }

  setVariable(key, value) {
    try {
      let variablesData = {};

      // Read existing variables if file exists
      try {
        variablesData = JSON.parse(fs.readFileSync("variables.json", "utf8"));
      } catch (error) {
        // File doesn't exist, use empty object
      }

      // Set the new variable
      variablesData[key] = value;
      variablesData["__updated__"] = new Date().toISOString();

      // Write back to file
      fs.writeFileSync(
        "variables.json",
        JSON.stringify(variablesData, null, 2),
      );

      return true;
    } catch (error) {
      console.error(`Error setting variable ${key}:`, error.message);
      return false;
    }
  }
}

// Create and export cronium instance
const croniumInstance = new Cronium();

// Make cronium globally available
global.cronium = croniumInstance;

module.exports = croniumInstance;

File: docker/executor/runtime-helpers/cronium.py
import json
import os
import sys

class Cronium:
    def __init__(self):
        # Load input data automatically
        try:
            with open("input.json", "r") as f:
                self._input_data = json.load(f)
        except FileNotFoundError:
            self._input_data = {}
        except json.JSONDecodeError:
            self._input_data = {}
        
        # Load event data automatically
        try:
            with open("event.json", "r") as f:
                self._event_data = json.load(f)
        except FileNotFoundError:
            self._event_data = {}
        except json.JSONDecodeError:
            self._event_data = {}
        
        # Initialize condition state
        self._condition = None

    def input(self):
        """Get input data"""
        return self._input_data

    def output(self, data):
        """Set output data to output.json file"""
        try:
            with open("output.json", "w") as f:
                json.dump(data, f, indent=2)
        except Exception as e:
            print(f"Error writing output: {e}", file=sys.stderr)

    def event(self):
        """Get event metadata"""
        return self._event_data

    def setCondition(self, condition):
        """Set a condition flag that can be checked by conditional events"""
        # Convert the condition to a boolean and store it
        self._condition = bool(condition)
        
        # Write the condition result to a file for the executor to read
        try:
            with open("condition.json", "w") as f:
                json.dump({"condition": self._condition}, f, indent=2)
        except Exception as e:
            print(f"Error writing condition: {e}", file=sys.stderr)
        
        return self._condition

    def getCondition(self):
        """Get the current condition state"""
        return self._condition

    def getVariable(self, key):
        """Get a variable value by key"""
        try:
            with open("variables.json", "r") as f:
                variables_data = json.load(f)
            return variables_data.get(key)
        except (FileNotFoundError, json.JSONDecodeError):
            return None

    def setVariable(self, key, value):
        """Set a variable value"""
        try:
            # Read existing variables if file exists
            try:
                with open("variables.json", "r") as f:
                    variables_data = json.load(f)
            except (FileNotFoundError, json.JSONDecodeError):
                variables_data = {}
            
            # Set the new variable
            variables_data[key] = value
            from datetime import datetime
            variables_data['__updated__'] = datetime.now().isoformat()
            
            # Write back to file
            with open("variables.json", "w") as f:
                json.dump(variables_data, f, indent=2)
            
            return True
        except Exception as e:
            print(f"Error setting variable {key}: {e}", file=sys.stderr)
            return False

# Create global cronium instance
_cronium_instance = Cronium()

# Export functions at module level for clean API
input = _cronium_instance.input
output = _cronium_instance.output
event = _cronium_instance.event
setCondition = _cronium_instance.setCondition
getCondition = _cronium_instance.getCondition
getVariable = _cronium_instance.getVariable
setVariable = _cronium_instance.setVariable

# Also provide the class for compatibility
cronium = _cronium_instance